### ISSUE TITLE
Anonymous inner function doesn't lint indent correctly for allman style

### ISSUE THREAD

**Issue by robinzimmermann** [eslint/eslint#10689]
**Tell us about your environment**

* **ESLint Version:** v3.12.2
* **Node Version:** v10.4.0
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**

Default, with ecmaVersion 6.

**Please show your full configuration:**

```json
{
  "parserOptions": {
    "ecmaVersion": 6
  },
  "rules": {
    "block-scoped-var": 1,
    "block-spacing": [ 1, "always" ],
    "brace-style": [ 1, "allman", { "allowSingleLine": true }  ],
    "camelcase": [ 1, { "properties": "always" } ],
    "comma-style": [ 1, "last" ],
    "curly": [ 1, "all" ],
    "default-case": 1,
    "eqeqeq": [ 1, "always" ],
    "func-style": [ 1, "expression" ],
    "indent": [ 1, 4, { "SwitchCase": 1, "MemberExpression": 1 } ],
    "key-spacing": [1, {"beforeColon": false, "afterColon": true, "mode": "minimum"}],
    "keyword-spacing": 1,
    "new-cap": 1,
    "no-alert": 1,
    "no-const-assign": 1,
    "no-empty": 1,
    "no-eq-null": 1,
    "no-extra-semi": 1,
    "no-fallthrough": 1,
    "no-floating-decimal": 1,
    "no-redeclare": 1,
    "no-multi-spaces": [ 1, {"exceptions": { "Property": true } } ],
    "no-nested-ternary": 1,
    "no-new": 1,
    "no-obj-calls": 1,
    "no-undef": 1,
    "no-undefined": 1,
    "no-unused-vars": 1,
    "no-use-before-define": 1,
    "no-mixed-spaces-and-tabs": 1,
    "no-shadow": 1,
    "no-tabs": 1,
    "no-trailing-spaces": 1,
    "no-unreachable": 1,
    "no-var": 1,
    "prefer-const": 1,
    "semi": 1,
    "semi-spacing": 1,
    "space-before-blocks": 1,
    "space-before-function-paren": [ 1, "never" ],
    "space-infix-ops": 1,
    "space-unary-ops": 1,
    "spaced-comment": [ 1, "always", { "exceptions": [ "-" ] } ],
    "valid-jsdoc": [ 1, { "requireReturn": false, "prefer": { "return": "returns" } } ]
  },
  "globals":{
    "jQuery": false,
    "$": true
  },
  "plugins": [
    "json"
  ]
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Here is the code snippet that triggers the issue:

```js
/* global console */

$('#some-element').click(function(event)
    {
        console.log(event.currentTarget);
    });
```

To run ESLint, I am using [gulp-eslint](https://www.npmjs.com/package/gulp-eslint) to invoke ESLint from within Gulp.

**What did you expect to happen?**

I expected the code snippet, above, to lint successfully.

That's because the braces and code block appy to the inner anonymous function, not the parent function.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  4:1  warning  Expected indentation of 0 spaces but found 4  indent
  5:1  warning  Expected indentation of 4 spaces but found 8  indent
  6:1  warning  Expected indentation of 0 spaces but found 4  indent
```

**Comment by not-an-aardvark** [comment]
Hi,

Thanks for the report. However, this is working as intended -- when using Allman style, the `indent` rule expects opening curly braces to have the same indentation as the function that opens them. In this case, it expects the `{` to have an indentation of 0 spaces because the `function(event)` text is on a line indented with 0 spaces.

**Comment by robinzimmermann** [comment]
Thanks for the response.
